app/angular/src/server/__tests__/angular-cli_config.test.ts

```diff
@@ -51,12 +51,23 @@ describe('angular-cli_config', () => {
     });
   });
 
+  it('should return null if `architect.build` option are not exists.', () => {
+    const angularJson = fs.readFileSync(path.resolve(__dirname, 'angular.json'), 'utf8');
+    const angularJsonWithNoBuildOptions = JSON.parse(stripJsonComments(angularJson));
+    angularJsonWithNoBuildOptions.projects['angular-cli'].architect.build = undefined;
+
+    getLeadingAngularCliProject(angularJsonWithNoBuildOptions);
+
+    const config = getAngularCliWebpackConfigOptions('/');
+    expect(config).toBeNull();
+  });
+
   it('should return baseConfig if no angular.json was found', () => {
     const baseConfig = { test: 'config' };
     const projectConfig = getAngularCliWebpackConfigOptions('test-path' as Path);
     const config = applyAngularCliWebpackConfig(baseConfig, projectConfig);
 
-    expect(projectConfig).toBe(undefined);
+    expect(projectConfig).toBe(null);
```

Comment:
Why `null` instead of undefined? `undefined` is nothing / just a declaration without a value, null is a value
That's why I changed it in another function here https://github.com/storybookjs/storybook/pull/8727/files

I don't want to be religious about that. In most cases it's not a problem up until you treat null as a valid value for something when it also can be undefined `const returnResult: SomeInterface | null | undefined`

Ignore this comment if you feel confident about this 👍